### ISSUE TITLE
Optional semicolon

### ISSUE THREAD

**Issue by ogoffart** [sixtyfpsui/sixtyfps#124]
Should we make the semicolon optional?

**Comment by tronical** [comment]
Yes, at the end of bindings at least, not inside code blocks IMO.

**Comment by appetrosyan** [comment]
My two cents would be that in QML, the separators got annoying. Due to the nature of the language where most stuff is essentially a dictionary, these separators are just redundant. 

Rather than making them optional (and causing a best practice where people still want you to use them), I’d consider a move similar to the one made for the algol-style binding operator (:=), deprecating it and making the cleaner syntax the new normal.

**Comment by rinie** [comment]
IMHO the 'Automatic Semicolon Insertion (ASI)' of Javascript is one of the most annoying 'features'.
Of course eslint airbnb can fix this, but what Javascript style to choose gives endless discussions.

Basic C like languages should use {}, semicolons as terminators and no significant whitespace outside of strings...
I do like trailing comma's...

Your focus on Rust would suggest keeping in line with the Rust style...

**Comment by ogoffart** [comment]
No plans to make the semicolon optional.

**Comment by appetrosyan** [comment]
Just a technicality, could you close this as not-planned as opposed to completed. This helps if someone comes across this discussion.
